### ISSUE TITLE
test failures on master

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#272]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/272) by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

I’m getting a failure with the below error message for the tests on python 3.12 only, you can see them in this pipeline. Technically that branch isn’t master, but it has only one trivial change far from this failure, so I believe it happens on master. I apparently don’t have write permission so I can’t start the test on master to confirm. 

‌

Here is the link to the failure: [https://bitbucket.org/joe\_fowler/mass/pipelines/results/1367/steps/\{a794ffd7-7853-45fd-a22d-41550e49627d\}](https://bitbucket.org/joe_fowler/mass/pipelines/results/1367/steps/%7Ba794ffd7-7853-45fd-a22d-41550e49627d%7D)

```
E   OSError: Unable to synchronously create file (unable to truncate a file which is already open)
```

‌

Also I think we’ve seen this error message before, but I failed to find it searching issues.

**Comment by joefowler** [comment]
@ggggggggg Is this error magically fixed by the move to GitHub and its Actions workflow?

**Comment by ggggggggg** [comment]
It seems to be intermittent, it wasn't persistent on bitbucket either.

**Comment by joefowler** [comment]
I saw it on my computer today. I suspect it has to do with `pytest` running some tests in parallel, but they try to write to the same HDF5 scratch file? Just a guess. Or if not in parallel, somehow a test isn't closing the HDF5 in time for the next to run successfully?

**Comment by joefowler** [comment]
Fixed by #284.
